src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -255,27 +255,83 @@ public void Clear()
         /// <returns>true if item contained; false if not</returns>
         public bool Contains(T item)
         {
-            if (_buckets != null)
+            int[]? buckets = _buckets;
+
+            if (buckets != null)
             {
                 int collisionCount = 0;
-                int hashCode = InternalGetHashCode(item);
                 Slot[] slots = _slots;
-                // see note at "HashSet" level describing why "- 1" appears in for loop
-                for (int i = _buckets[hashCode % _buckets.Length] - 1; i >= 0; i = slots[i].next)
+
+                IEqualityComparer<T>? comparer = _comparer;
+
+                if (comparer == null)
                 {
-                    if (slots[i].hashCode == hashCode && _comparer.Equals(slots[i].value, item))
+                    int hashCode = item == null ? 0 : InternalGetHashCode(item.GetHashCode());
+
+                    if (default(T)! != null) // TODO-NULLABLE: default(T) == null warning (https://github.com/dotnet/roslyn/issues/34757)
```

Comment:
Let's remove this trailing comment.

Reply:
@safern, I don't have a strong opinion, but why do you want to remove the comment?  We still have this in a bunch of places in coreclr... do you want to remove it from all of those (separate from this PR)?

Reply:
This was a new line added in this PR, I was suggesting to not include a comment based on past discussion here: https://github.com/dotnet/corefx/pull/40651#discussion_r320524371 -- but I don't have a strong opinion on this either.

If we'd like to remove all the comments and then address the removal of `!` later I can happily submit a PR.